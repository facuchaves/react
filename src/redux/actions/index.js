export const ACTIONS = {
  UPDATE_SEARCH_INPUT: "UPDATE_SEARCH_INPUT",
  UPDATE_RATING: "UPDATE_RATING",
  CLICK_PLAYER: "CLICK_PLAYER",
  SEARCH_ISSUES: "SEARCH_ISSUES",
};

export const updateSearchInput = (searchInput) => {
  return {
    type: ACTIONS.UPDATE_SEARCH_INPUT,
    payload: searchInput,
  };
};

export const updateRating = (rating) => {
  return {
    type: ACTIONS.UPDATE_RATING,
    payload: rating,
  };
};

export const clickPlayer = () => {
  return {
    type: ACTIONS.CLICK_PLAYER,
  };
};

export const searchIssues = ( query ) => {
  console.log("Action")
  console.log("query: ",query)
  return {
    type: ACTIONS.SEARCH_ISSUES,
    payload: query,
  };
};