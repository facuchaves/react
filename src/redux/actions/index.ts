export const ACTIONS = {
  SEARCH_ENTITIES: 'SEARCH_ENTITIES',
};

export const searchEntities = (query: any) => ({
  type: ACTIONS.SEARCH_ENTITIES,
  payload: {
    query,
  },
});
