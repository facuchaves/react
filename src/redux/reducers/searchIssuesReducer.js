import { ACTIONS } from "../actions";

const INITIAL_STATE = {
  q:"",
  status:"OPEN"
};

const searchIssuesReducer = (state = INITIAL_STATE, action) => {
  console.log("reducer")
  console.log("action: ",action)
  console.log("state: ",state)
  switch (action.type) {
    case ACTIONS.SEARCH_ISSUES:
      return {
        ...state,
        query: action.payload,
      };
    default:
      return state;
  }
};

export default searchIssuesReducer;
