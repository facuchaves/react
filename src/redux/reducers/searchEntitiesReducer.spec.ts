import {ACTIONS} from '../actions';
import searchIssuesReducer from './searchEntitiesReducer';

test('searchIssuesReducer', () => {
  const state = {
    query: {
      repo: 'facebook/react',
      q: '',
      state: 'OPEN',
    },
  };

  const payload = {
    query: {
      q: 'Query',
      state: 'CLOSED',
    },
  };

  expect(
    searchIssuesReducer(state, {type: ACTIONS.SEARCH_ENTITIES, payload}),
  ).toMatchSnapshot();
});
