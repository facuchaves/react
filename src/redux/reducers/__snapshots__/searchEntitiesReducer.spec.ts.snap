// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchIssuesReducer 1`] = `
Object {
  "query": Object {
    "q": "Query",
    "repo": "facebook/react",
    "state": "CLOSED",
  },
}
`;
